### PR TITLE
[MM-46448] - Subscribe button for Professional is disabled during Enterprise trial

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
When on enterprise trial, enable professional upgrade button

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-46448

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by neallred — APPROVED** [review]

File: e2e/cypress/tests/integration/enterprise/cloud/billing/cloud_pricing_modal_spec.js
Comment (marianunez) [inline e2e/cypress/tests/integration/enterprise/cloud/billing/cloud_pricing_modal_spec.js:276]: Can we keep the case of disabled if they are indeed in Enterprise (no trial)?

**Comment by marianunez** [comment]
Adding @stevemudie due to timezone - we are hoping to cherry pick for next cloud release

**Comment by stevemudie** [comment]
This worked as expected so it's all set. It just needs to run through an E2E before I can approve it.

I'm only waiting to do so because of what @marianunez [mentioned above](https://github.com/mattermost/mattermost-webapp/pull/10964#discussion_r950554205).

File: e2e/cypress/tests/integration/enterprise/cloud/billing/cloud_pricing_modal_spec.js
Comment (AGMETEOR) [reply e2e/cypress/tests/integration/enterprise/cloud/billing/cloud_pricing_modal_spec.js:276]: That test case is below on line 434

**Review by marianunez — APPROVED** [review]

**Comment by marianunez** [comment]
@stevemudie @yasserfaraazkhan can we prioritize this QA Review? We would like to cherry pick this to cloud today for this week's release

**Review by yasserfaraazkhan — APPROVED** [review]
@AGMETEOR , I tested it local, and it is working as expected

**Comment by marianunez** [comment]
Thanks @yasserfaraazkhan!

**Comment by marianunez** [comment]
@amyblais heads up that the automated cherry pick to the Milestone once merged is failing for `cloud` given that it's trying to do the branch `release-cloud` instead of `cloud`. Wondering if this something the release team can help with?

**Comment by amyblais** [comment]
> @amyblais heads up that the automated cherry pick to the Milestone once merged is failing for `cloud` given that it's trying to do the branch `release-cloud` instead of `cloud`. Wondering if this something the release team can help with?

Sure, I opened a ticket: https://mattermost.atlassian.net/browse/CLD-4001.

**Pull request merged**
